lib/ethpy/ethpy/hyperdrive/api/api_test.py

```diff
@@ -23,102 +25,208 @@
 class TestHyperdriveInterface:
     """Tests for the HyperdriveInterface api class."""
 
+    def setup_hyperdrive_interface(self, _local_hyperdrive_pool: DeployedHyperdrivePool) -> HyperdriveInterface:
+        """Set up the hyperdrive interface to access a deployed hyperdrive pool.
+
+        All arguments are fixtures.
+
+        Returns
+        -------
+            HyperdriveInterface
+                The interface to access the deployed hyperdrive pool.
+        """
+        rpc_uri = cast(HTTPProvider, _local_hyperdrive_pool.web3.provider).endpoint_uri or URI("http://localhost:8545")
+        hyperdrive_contract_addresses: HyperdriveAddresses = _local_hyperdrive_pool.hyperdrive_contract_addresses
+        eth_config = EthConfig(artifacts_uri="not used", rpc_uri=rpc_uri, abi_dir="./packages/hyperdrive/src/abis")
+        return HyperdriveInterface(eth_config, addresses=hyperdrive_contract_addresses)
+        # TODO: need an agent address to mock up trades
+
     def test_pool_config(self, local_hyperdrive_pool: DeployedHyperdrivePool):
         """Checks that the Hyperdrive pool_config matches what is returned from the smart contract.
 
         All arguments are fixtures.
         """
-        uri: URI | None = cast(HTTPProvider, local_hyperdrive_pool.web3.provider).endpoint_uri
-        rpc_uri = uri if uri else URI("http://localhost:8545")
-        abi_dir = "./packages/hyperdrive/src/abis"
-        hyperdrive_contract_addresses: HyperdriveAddresses = local_hyperdrive_pool.hyperdrive_contract_addresses
-        eth_config = EthConfig(artifacts_uri="not used", rpc_uri=rpc_uri, abi_dir=abi_dir)
-        hyperdrive = HyperdriveInterface(eth_config, addresses=hyperdrive_contract_addresses)
         # TODO: remove cast when pypechain consolidates dataclasses.
```

Comment:
you can remove this TODO it's resolved

Reply:
done in 9fee6d551c142a93c069027ca22679f3cd037182